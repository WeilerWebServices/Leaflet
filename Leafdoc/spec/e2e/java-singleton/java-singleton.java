// Adapted from https://en.wikibooks.org/wiki/Computer_Science_Design_Patterns/Singleton

/**
 * 🍂class SingletonObject
 * Generic implementation of a singleton
 */
class SingletonObject {
	// 🍂section Private
	// 🍂property SingletonObject: object; The insternal instance of the object.
	private static SingletonObject object;

	// 🍂method SingletonObject(): null; Instantiates the object, only called once.
	private SingletonObject() {
		// Instantiate the object.
	}

	// 🍂section Public
	// 🍂method getInstance(): SingletonObject; Returns the singleton instance.
	public static SingletonObject getInstance() {
		if (object == null) {
			object = new SingletonObject(); // Create the object for the first and last time
		}
		return object;
	}
}
