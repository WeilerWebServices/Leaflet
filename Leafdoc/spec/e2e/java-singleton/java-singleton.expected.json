{
 "SingletonObject": {
  "name": "SingletonObject",
  "aka": [],
  "comments": [
   "Generic implementation of a singleton"
  ],
  "supersections": {
   "property": {
    "name": "property",
    "aka": [],
    "comments": [],
    "sections": {
     "Private": {
      "name": "Private",
      "aka": [],
      "comments": [],
      "uninheritable": false,
      "documentables": {
       "SingletonObject": {
        "name": "SingletonObject",
        "aka": [],
        "comments": [
         "The insternal instance of the object."
        ],
        "params": {},
        "type": "object",
        "optional": false,
        "defaultValue": null,
        "id": "singletonobject-singletonobject"
       }
      },
      "type": "property",
      "id": "singletonobject-private"
     }
    },
    "id": "singletonobject-property"
   },
   "method": {
    "name": "method",
    "aka": [],
    "comments": [],
    "sections": {
     "Private": {
      "name": "Private",
      "aka": [],
      "comments": [],
      "uninheritable": false,
      "documentables": {
       "SingletonObject": {
        "name": "SingletonObject",
        "aka": [],
        "comments": [
         "Instantiates the object, only called once."
        ],
        "params": {},
        "type": "null",
        "optional": false,
        "defaultValue": null,
        "id": "singletonobject-singletonobject"
       }
      },
      "type": "method",
      "id": "singletonobject-private"
     },
     "Public": {
      "name": "Public",
      "aka": [],
      "comments": [],
      "uninheritable": false,
      "documentables": {
       "getInstance": {
        "name": "getInstance",
        "aka": [],
        "comments": [
         "Returns the singleton instance."
        ],
        "params": {},
        "type": "SingletonObject",
        "optional": false,
        "defaultValue": null,
        "id": "singletonobject-getinstance"
       }
      },
      "type": "method",
      "id": "singletonobject-public"
     }
    },
    "id": "singletonobject-method"
   }
  },
  "inherits": [],
  "relationships": [],
  "id": "singletonobject"
 }
}